syntax = "proto3";
package common;

service ReplicationService {
    rpc ReplicateMessage(ReplicationRequest) returns (ReplicationResponse);
}

message ReplicationRequest {
    string message_id = 1;
    string message_body = 2;
}

message ReplicationResponse {
    Status status = 1;
}

enum Status {
    STATUS_UNSPECIFIED = 0;
    SUCCESS = 1;
    FAILURE = 2;
}